server/plugin.go

```diff
@@ -142,6 +142,22 @@ func (p *Plugin) GetClientForUser(userID string) (msteams.Client, error) {
 	if token == nil {
 		return nil, errors.New("not connected user")
 	}
+
+	if token.Expiry.Before(time.Now()) {
```

Comment:
Should we more aggressive with our window handling? For example, if the token hasn't expired yet, but does so in 1 second, by the time I use this client it might have expired, the client will silently renew it, and we won't ever save that back to the database.

Reply:
This can happen, but whenever we get the token again from the database is going to be obsolete enough, and eventually we are going to reach the point where the background job that keeps them alive is going to detect that the session is not properly updated and refresh it anyway. We need to have very bad luck to reach that point, but even with bad luck, that point should take care of it.

Reply:
Done